### ISSUE TITLE
Footer in website 

### ISSUE THREAD

**Issue by YasirObaibullah99** [aditya-singh9/kekfinder#34]
I can add a footer to the website which will make the website more attractive and which will make a complete website

**Comment by Substancia** [comment]
Hey @YasirObaibullah99 is this a critical work, and if I proceed with https://github.com/aditya-singh9/kekfinder/issues/41, will it be breaking changes to your work?

**Comment by YasirObaibullah99** [comment]
No you just tell me how you are doing in #41 it will not break

**Comment by Substancia** [comment]
Sure. Basically the "components" folder will have only folders, each folder named different components (like items, search etc), and each folder will have the component files (js) and their supporting files like css (basically all files with same name goes in sane folder with that name).

If you're creating new component files, please follow this practice. I'll take care of converting the structure of existing files.

**Comment by Substancia** [comment]
Reference: https://github.com/Substancia/OneshotAI-MERN-client-webapp/

**Comment by YasirObaibullah99** [comment]
Ya this is best practices for react js and go for it

**Comment by Substancia** [comment]
@YasirObaibullah99 is this work done? Let me know once I can start with the code restructuring without affecting your work. Thanks.

**Comment by YasirObaibullah99** [comment]
Ya it merged now you can start your work

**Comment by Substancia** [comment]
Awesome!

@aditya-singh9 @Rudresh-pandey since the [new UI work](https://github.com/aditya-singh9/kekfinder/issues/46) starts tomorrow and the [repo will be blocked](https://github.com/aditya-singh9/kekfinder/issues/46#issuecomment-1265693410) for the entire day, and since no critical work is being done right now, I'll do the code restructuring after my dinner. That cool?

**Comment by aditya-singh9** [comment]
Cool

**Comment by Substancia** [comment]
Why are we using an entire library (cdbreact) just for the footer? Couldn't we have just coded a footer component manually? An additional package means higher vulnerability risk in the future.

Also the footer is [misbehaving](https://github.com/aditya-singh9/kekfinder/pull/70#issuecomment-1268308419).

**Comment by Rudresh-pandey** [comment]
left-bottom in **desktop view**
bottom in **tablet and mobile view**

**Comment by Rudresh-pandey** [comment]
i am not getting any of the footer issue in my local npm server

**Comment by Substancia** [comment]
Sorry, my bad, I've got this particular issue [fixed](https://github.com/aditya-singh9/kekfinder/pull/70#issuecomment-1268430411), with a new question. Do we want the footer to stick to the bottom when emojis list is empty? Because that wasn't implemented.

**Comment by Rudresh-pandey** [comment]
![image](https://user-images.githubusercontent.com/96531798/194073037-0b29ca8a-5658-44fe-bd96-833d58aa3661.png)
i have already implemented it

**Comment by Rudresh-pandey** [comment]
![image](https://user-images.githubusercontent.com/96531798/194074321-9393fc32-a0e1-42fe-b330-7b87953a6a21.png)

**Comment by Rudresh-pandey** [comment]
check this :- it was working when i made the pr 
https://kekfinder-cnjf9wcez-aditya-singh9.vercel.app/

**Comment by Rudresh-pandey** [comment]
ok now i understand , since i was using my laptop to see the progress i didn't check the height of the device that's why the footer is behaving like this , it's not a big issue

**Comment by Rudresh-pandey** [comment]
i am just confused 😑, i don't know which property should i use to make this footer on the bottom of the screen

**Comment by Rudresh-pandey** [comment]
> Sorry, my bad, I've got this particular issue [fixed](https://github.com/aditya-singh9/kekfinder/pull/70#issuecomment-1268430411), with a new question. Do we want the footer to stick to the bottom when emojis list is empty? Because that wasn't implemented.

yes i didn't considered the heights of the screen , i wasted a lot of time 😑 and didn't allow others to get there code merged

**Comment by Substancia** [comment]
I did spend some time on this issue, and realized that the difficulty in this issue arises from the way the divisions and blocks are defined in the html code (basically the elements structure). To fix this, all I see are either do some nasty hacks (like force fixing heights of certain elements), or restructuring the elements altogether. If anyone can come up with a GOOD solution, I'll be very curious.

**Comment by Substancia** [comment]
[This](https://github.com/aditya-singh9/kekfinder/pull/73) is what I came up with, simple change, works fine. Please do some edge-testing.
@aditya-singh9 @Rudresh-pandey
